enterprise/server/remote_execution/filecache/filecache.go

```diff
@@ -196,15 +202,28 @@ func (c *fileCache) scanDir() {
 	close(c.dirScanDone)
 }
 
-func key(node *repb.FileNode) string {
+func groupSpecificKey(groupID string, node *repb.FileNode) string {
 	suffix := ""
 	if node.GetIsExecutable() {
 		suffix = "." + executableSuffix
 	}
-	return node.GetDigest().GetHash() + suffix
+	return groupID + "/" + node.GetDigest().GetHash() + suffix
+}
+
+func groupIDStringFromContext(ctx context.Context) string {
+	if c, err := claims.ClaimsFromContext(ctx); err == nil {
+		return strings.ToUpper(c.GroupID)
```

Comment:
nit: group IDs are always uppercase, not sure why this is needed?

Reply:
Done.